### PR TITLE
[Model Runner V2] Spec decode rejection sampler greedy support

### PR DESCRIPTION
**Pull request by TheEpicDolphin**
# Purpose
Following up on https://github.com/vllm-project/vllm/pull/35461, specifically with support for greedy sampling (temperature = 0).

In order to support this in an efficient way, I get local argmax/max from target logits for greedy requests in `_gather_draft_logits_and_target_argmax_kernel`. Then during `_probabilistic_rejection_kernel`, the target argmax token is sampled only for the greedy requests. This limits the performance impact of greedy requests on the rest of the batch.

### CODE REVIEW

File: vllm/v1/worker/gpu/spec_decode/rejection_sampler.py [hunk 1]
```diff
@@ -68,202 +68,338 @@ def strict_rejection_sample(
...
     # [num_reqs, num_speculative_steps + 1]
-    sampled = draft_sampled.new_empty(
+    sampled = draft_sampled.new_zeros(
         num_reqs, num_speculative_steps + 1, dtype=torch.int64
     )
     # [num_reqs]
     rejected_steps = sampled.new_empty(num_reqs)
-    _probabilistic_rejection_sample_kernel[(num_reqs,)](
+    # [num_reqs]
+    rejected_pos = pos.new_empty(num_reqs)
```
Comment (TheEpicDolphin) [inline vllm/v1/worker/gpu/spec_decode/rejection_sampler.py:361]: I felt it made more sense to compute this in `_probabilistic_rejection_kernel` rather than `_compute_residual_logits_kernel`, so i moved it here. Also, renamed it from `residual_pos` to `rejected_pos`.

**Review by WoosukKwon — APPROVED** [review]
Thanks for the PR!

I think we can fuse more kernels to minimize the materialization of `*_logits` tensors, but we can probably follow up after this.

**Comment by geraldstanje1** [comment]
hi @TheEpicDolphin what gpu was used in this test and what vllm settings regarding max_batched_tokens etc? also have you done any tests with prefix caching enabled?

**Comment by TheEpicDolphin** [comment]
@geraldstanje1 i used H200, and I didn't have the chance to test with max_batched_tokens or prefix caching.

Also please ignore the benchmark results from this PR, there was a bug skewing the acceptance rates that i fixed later. [This PR](https://github.com/vllm-project/vllm/pull/38496) has the most recent benchmark results for rejection sample :)

**Comment by geraldstanje1** [comment]
@TheEpicDolphin can you show how you run those benchmarks in https://github.com/vllm-project/vllm/pull/38496 - i assume you also have speculative decoding enabled?

**Comment by TheEpicDolphin** [comment]
@geraldstanje1 yep, i used spec decoding and compared strict vs probabilistic rejection sampling methods. I added the server/benchmark commands to https://github.com/vllm-project/vllm/pull/38496
